### ISSUE TITLE
DAR 401 false positive for exception caught within function

### ISSUE THREAD

**Issue by dmartindelorme** [terrencepreilly/darglint#93]
With darglint v1.3.0, and for a function such as:
```python
def test(a: int, b: int) -> int:
    """Sums a and b.

    Args:
        a: First integer.
        b: Second integer.

    Returns:
        The sum.
    """
    try:
        raise ValueError("always")
        return a + b
    except Exception as e:
        print(e)
```
darglint returns:
```bash
DAR401: Missing exception(s) in Raises section: -r ValueError
```

Do you think it's going to be fixed in v1.3.1 along with #88 and #68?

**Comment by terrencepreilly** [comment]
This won't be fixed in v1.3.1, though I've been thinking about doing this for a while.  (Apparently for [three years](https://github.com/terrencepreilly/darglint/blame/1cdb8f51f4341884001836a67eaa59318487150b/README.md#L427)).

In the general case, I don't think it's very feasible, because it would require knowing the superclass of any given error.  That sort of analysis is out of scope for darglint. (It would require loading/analyzing an entire project and its dependencies.)

There's maybe some subclass of the problem which could be tackled. For example, we could probably assume `Exception` is a superclass of anything which can be thrown.  Unfortunately, I don't think that's true: all exceptions have to derive from `BaseException`, even though all user-defined exceptions should subclass `Exception`.  For example, the following still produces an exception:

```
class CustomException(BaseException):
    pass

try:
    raise CustomException()
except Exception:
    print('Never executed.')
```

So, short of running an entire project and its dependencies, it's not clear to me how this could be done.  Do you agree, or have any suggestions?
